neural/neural_discern_run_script.py

```diff
@@ -0,0 +1,342 @@
+        accuracy_df = pd.concat([accuracy_df, accuracy], sort=True)
+
+    return micro_f1_df, macro_f1_df, accuracy_df
+
+
+def highlight_attnw_over_sents(docid_attnweights_map, proc_articles_repr, topk=5):
+    for docid in docid_attnweights_map:
+        attnw = docid_attnweights_map[docid]
+        topk = topk if attnw.size(-1) > topk else attnw.size(-1)  # get top
+        max_val, max_indx = torch.topk(attnw, topk, dim=1)
+        print(docid)
+        print("attended sent:")
+        for i in range(max_indx.size(-1)):
+            target_indx = max_indx[0][i].item()
+            print("sentence num:", target_indx, "attnw:", max_val[0][i].item())
+            print(proc_articles_repr[docid]['sents'][target_indx])
+        print()
+
+
+def verbose_print(text, print_yn):
+    if print_yn:
+        print(text)
+
+# ============================================================
+
+
+if __name__ == '__main__':
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--test-mode", action="store_true", default=False, help="run in abbreviated test mode")
+    parser.add_argument("--biobert", default=True, help="Use the BioBERT embeddings, or the normal BERT")
+    parser.add_argument("--rewrite-sentence-embeddings", action="store_true", default=False, help="(Re-)compute the "
+                                                                                                  "sentence embeddings")
+    parser.add_argument("--run-hyper-param-search", default=True, help="Run the hyper parameter search")
+    parser.add_argument("--hyperparam-search-dir", default=None, help="Re-use the results of a pre-run hyperparam "
+                                                                      "search")
+    parser.add_argument("--base-dir", default='/opt/data/autodiscern/aa_neural', help="Base dir to and including "
+                                                                                      "autodiscern/aa_neural/")
+    args = parser.parse_args()
+
+    config = {
+        'test_mode': args.test_mode,
+        'biobert': args.biobert,
+        'rewrite_sentence_embeddings': args.rewrite_sentence_embeddings,
+        'run_hyper_param_search': args.run_hyper_param_search,
+        'hyperparam_search_dir': args.hyperparam_search_dir,
+        'questions_to_run': [4, 5, 9, 10, 11],
+        'max_folds': 5,
+        'num_epochs': 25,
+        'verbose': True,
+        'experiment_to_rerun': None,
+        'questions': (4, 5, 9, 10, 11),
+        'question_gpu_map': {4: 1, 5: 2, 9: 3, 10: 4, 11: 5},
+        'base_dir': args.base_dir,
+    }
+
+    if config['hyperparam_search_dir'] and config['run_hyper_param_search']:
+        print("WARNING: you selected a hyperparam search dir while also setting run-hyper-param-search as True. "
+              "The pre-built hyperparam search dir will be used. Hyperparam search will not be run.")
+
+    if config['test_mode']:
+        config['max_folds'] = 2
+        config['num_epochs'] = 2
```

Comment:
By default it should run only once (i.e. one pass over the test data)

Reply:
I think you're thinking of the prediction script? These are a set of configs that are used in "test mode" for faster testing/debugging of the model training script - only run 2 partitions instead of all 5, only run 2 epochs instead of 25. Otherwise, the num_epochs for training defaults to 25.

Reply:
Oh I see -- it refers to `testing mode` (pipeline perspective) rather than model testing.